### PR TITLE
Add modularity example to Wiki

### LINKED ISSUE #176: Create Wiki docs page - [Variables] custom_policy_roles

### LINKED ISSUE #392: Add documentation for deploying across multiple Terraform workspaces (Terraform state file segregation)
<h3 id=communitynote>Community Note </h3>
<ul>
<li>Please vote on this issue by adding a &#128077; <a href="https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/">reaction</a> to the original issue to help the community and maintainers prioritize this request </li>
<li>Please do not leave &quot;+1&quot; or &quot;me too&quot; comments, they generate extra noise for issue followers and do not help prioritize the request </li>
<li>If you are interested in working on this issue or have submitted a pull request, please leave a comment </li>
</ul>
<h3 id=description>Description </h3>
<p><strong>Is your feature request related to a problem?</strong> </p>
<p>No, this is a documentation request. </p>
<p><strong>Describe the solution you'd like</strong> </p>
<p>As described in #391, some customers would like to know how to implement the platform components of ALZ using multiple instances of this module to allow segregation of resources across multiple Terraform workspaces. </p>
<p>This helps to reduce the number of resources within a single Terraform state file, but also allows more granular RBAC to be applied to pipelines. </p>
<p>For example, a common scenario might be to have the following workspaces configured: </p>
<p>| Workspace / Module | Description |
| :--- | :--- |
| alz<em>core | Implementation of the core Management Group Hierarchy, including all base policies and IAM settings. This workspace will have high privileges across the resource hierarchy but can be under strict change control as changes are rare. |
| alz</em>landing<em>zones | Implementation of additional custom Management Groups under the <code>landing_zones</code> Management Group implemented by <code>alz_core</code>. Changes will be more common under this scope, as new landing zone Subscriptions are regularly added to the environment. This workspace can be run under an identity with permissions delegated to the <code>landing_zones</code> Management Group only, reducing attack surface area. |
| alz</em>connectivity | Implementation of the <code>connectivity</code> resources into the corresponding Azure subscription. Permissions for this workspace are only needed on the target Azure subscription, reducing attack surface area. |
| alz_management | Implementation of the <code>management</code> resources into the corresponding Azure subscription. Permissions for this workspace are only needed on the target Azure subscription, reducing attack surface area. | </p>
<p>This is already possible with the module, but documentation needs to be created so customers know how to implement this pattern (or similar) in their environments. </p>
<p>This issue is being raised to request and track addition of documentation for the above to the Wiki. </p>
<p><strong>Additional context</strong> </p>
<ul>
<li>Please refer to #391 </li>
</ul>

### LINKED ISSUE #567: Feature Request - Videos to Assist Written Documentation
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

Had request from customer that they would appreciate additional videos that could cover the following:

-  Videos that would walk through some of the [examples](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/wiki/Examples).
-  Videos that could assist with giving a deeper look at how the module is designed and correlates to the different configurations available.
- Videos talking through some of the limitations as well as some of the things to watch out for.

### PR DESCRIPTION
**Pull request by krowlandson**
## Overview/Summary

This PR adds Wiki documentation and a working example showing how to segregate deployment of an Azure landing zone across multiple module instances for connectivity, management and core resources.

## This PR fixes/adds/changes/removes

1. Close #392
2. Close [AB#24750](https://dev.azure.com/CSUSolEng/2a9d5a5a-8998-4ad0-81c8-ef3045e4da97/_workitems/edit/24750)
3. Close #176
4. Close #567

### Breaking Changes

None

## Testing Evidence

n/a

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated relevant and associated documentation.
- [x] Updated the ["RELEASE"](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/issues) page with details needed for when this code change is released.

### CODE REVIEW

File: docs/wiki/[Examples]-Deploy-using-multiple-module-declarations.md [hunk 1]
```diff
@@ -0,0 +1,806 @@
...
+  - [variables.tf](#variablestf)
+  - modules/
+    - connectivity/
+      - [main.tf](#modulesconnectivitymaintf)
+      - [settings.connectivity.tf](#modulesconnectivitysettingsconnectivitytf)
+      - [variables.tf](#modulesconnectivityvariablestf)
+    - core/
+      - [main.tf](#modulescoremaintf)
+      - [settings.core.tf](#modulescoresettingscoretf)
+      - [settings.core.tf](#modulescoresettingsidentitytf)
```
Comment (matt-FFFFFF) [inline docs/wiki/[Examples]-Deploy-using-multiple-module-declarations.md:41]: duplicate file name

**Review by matt-FFFFFF — COMMENTED** [review]
Just one minor point - otherwise great!

File: docs/wiki/[Examples]-Deploy-using-multiple-module-declarations.md (hunk 1 above)
Comment (krowlandson) [reply docs/wiki/[Examples]-Deploy-using-multiple-module-declarations.md]: Good spot... Should be resolved now.

Have also made some other updates so PR ready for review again please.

**Review by matt-FFFFFF — APPROVED** [review]

**Review by jtracey93 — APPROVED** [review]
LGTM

**Pull request merged**